lib/data_magic/document_builder.rb

```diff
@@ -0,0 +1,140 @@
+require './lib/expression'
+
+module DataMagic
+  module DocumentBuilder
+    class << self
+      def logger
+        DataMagic::Config.logger
+      end
+
+      # parse a row from a csv file, returns a nested document
+      # row: a hash  { field => value } where all values are strings
+      # fields: column_name => field_name
+      # config: DataMagic.Config instance for dictionary, column types, NULL
+      def parse_row(row, fields, config, options={}, additional=nil)
+        row = csv_row = row.to_hash
```

Comment:
this line scares me. I just hit a bug with this kind of multiple assignment the other day, because I forgot that changing one of the variables would change the other too. Is there a better way of doing this?